### PR TITLE
refactor logging

### PR DESCRIPTION
**Pull request by AChoffat**
First try to change logging logic for exceptions

### CODE REVIEW

File: django_socio_grpc/services/servicer_proxy.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ async def handler(request: Message, context) -> AsyncIterable[Message]:
                 request, proxy_context, action, service_instance
             )
             async for response in await safe_async_response(
-                self._middleware_chain, request_container, self.async_process_exception
+                self._middleware_chain, request_container
             ):
                 yield response.grpc_response
```
Comment (legau) [inline django_socio_grpc/services/servicer_proxy.py:187]: missing the streaming logs

File: django_socio_grpc/services/servicer_proxy.py [hunk 2]
```diff
@@ -264,32 +271,40 @@ def __getattr__(self, action):
...
+            if exception.logging_level == "WARNING":
+                request_logger.warning(
+                    f"{type(request_container.service).__name__}.{request_container.service.action} : {exception.status_code.name}",
+                    exc_info=exception,
+                )
+            if exception.logging_level == "ERROR":
+                request_logger.error(
+                    f"{type(request_container.service).__name__}.{request_container.service.action} : {exception.status_code.name}",
+                    exc_info=exception,
+                )
```
Comment (legau) [inline django_socio_grpc/services/servicer_proxy.py:305]: simplify this

File: django_socio_grpc/utils/utils.py [hunk 3]
```diff
@@ -22,20 +22,14 @@ async def async_generator():
                 async for item in response:
                     yield item
             except Exception as e:
-                if process_exceptions:
-                    await process_exceptions(e, request)
-                else:
-                    raise e
+                raise e
```
Comment (legau) [inline django_socio_grpc/utils/utils.py:25]: keep it

File: django_socio_grpc/services/servicer_proxy.py [hunk 4]
```diff
@@ -264,32 +271,40 @@ def __getattr__(self, action):
...
-                exc_info=exc,
-            )
-            await request_container.context.abort(exc.status_code, exc.get_full_details())
-        elif isinstance(exc, (grpc.RpcError, grpc.aio.AbortError)):
-            raise exc
+            await context.abort(exc.status_code, exc.get_full_details())
+        else:
+            await context.abort(grpc.StatusCode.UNKNOWN, str(exc))
+
+    def log_request_success(self, response, request_container):
```
Comment (legau) [inline django_socio_grpc/services/servicer_proxy.py:284]: As per https://docs.djangoproject.com/en/4.2/ref/logging/#django-request the logging of success response is in fact not necessary

**Review by legau — CHANGES_REQUESTED** [review]

File: django_socio_grpc/services/servicer_proxy.py (hunk 2 above)
Comment (AChoffat) [reply django_socio_grpc/services/servicer_proxy.py:303]: any idea?

Comment (legau) [reply django_socio_grpc/services/servicer_proxy.py]: The string is the same for all, and the logging level has the name of the function called on the logger

File: django_socio_grpc/utils/utils.py (hunk 3 above)
Comment (AChoffat) [reply django_socio_grpc/utils/utils.py]: We changed the calls to not use this anymore though. I put it back but it's not used anymore.

File: django_socio_grpc/services/servicer_proxy.py [hunk 5]
```diff
@@ -272,6 +284,7 @@ def __getattr__(self, action):
     def process_exception(self, exc, request_container: GRPCRequestContainer):
         if isinstance(exc, GRPCException):
             request_container.context.abort(exc.status_code, exc.get_full_details())
+            print(exc)
```
Comment (legau) [inline django_socio_grpc/services/servicer_proxy.py:287]: -

File: django_socio_grpc/services/servicer_proxy.py [hunk 6]
```diff
@@ -281,14 +294,20 @@ async def async_process_exception(self, exc, context):
...
             "status_code": request_container.context.code(),
         }
         path = f"{self.service_class.get_service_name()}/{request_container.action}"
-        message = f"{type(exception).__name__} : {path}"
-        if isinstance(exception, GRPCException):
-            exception.log_exception(request_logger, message, extra=extra)
+
+        if not exception:
+            if grpc_settings.LOG_OK_RESPONSE or settings.DEBUG:
+                message = f"Response OK: {path}"
```
Comment (legau) [inline django_socio_grpc/services/servicer_proxy.py:306]: OK : path

File: django_socio_grpc/services/servicer_proxy.py [hunk 7]
```diff
@@ -264,32 +283,30 @@ def __getattr__(self, action):
...
+    def log_response(self, exception, request_container):
+        extra = {
+            "request": request_container.grpc_request,
+            "status_code": request_container.context.code(),
+        }
+        path = f"{self.service_class.get_service_name()}/{request_container.action}"
+
+        if not exception:
+            if grpc_settings.LOG_OK_RESPONSE or settings.DEBUG:
+                message = f"OK: {path}"
```
Comment (legau) [inline django_socio_grpc/services/servicer_proxy.py:305]:
```suggestion
                message = f"OK : {path}"
```

File: django_socio_grpc/log.py [hunk 8]
```diff
@@ -1,175 +0,0 @@
-"""
-logging utils
-"""
-import asyncio
```
Comment (AMontagu) [inline django_socio_grpc/log.py:4]: deleting this file will create breaking change and crash to app still using it.
Are we sure we want that @legau ?

Comment (legau) [reply django_socio_grpc/log.py:4]: good time to do it imo

File: django_socio_grpc/services/servicer_proxy.py [hunk 9]
```diff
@@ -197,10 +203,15 @@ async def handler(request: Message, context) -> Awaitable[Message]:
             request_container = GRPCRequestContainer(
                 request, proxy_context, action, service_instance
             )
-            response = await safe_async_response(
-                self._middleware_chain, request_container, self.async_process_exception
-            )
-            return response.grpc_response
+            try:
+                exc = None
+                response = await safe_async_response(self._middleware_chain, request_container)
```
Comment (AMontagu) [inline django_socio_grpc/services/servicer_proxy.py:208]: safe_async_response look like it is not using process_exceptions arguments. Should be deleted then

File: django_socio_grpc/log.py (hunk 8 above)
Comment (AMontagu) [reply django_socio_grpc/log.py:4]: I don't understand why removing set_log_record_factory function as it add valudable context if needed in classique log. Come to see you to talk about that

**Pull request merged**